rs/consensus/src/consensus/validator.rs

```diff
@@ -3881,12 +3880,7 @@ pub mod test {
                 .ok();
 
             let changeset = validator.on_state_change(&PoolReader::new(&pool));
-            assert_matches!(
```

Comment:
@kpop-dfinity @dist1ll @eichhorl the expectation here is changed 
PTAL

Reply:
This test started to fail because the block time should've included the unit delay. But the expectation otherwise should not be changed. My suggestion:

```diff
diff --git a/rs/consensus/src/consensus/validator.rs b/rs/consensus/src/consensus/validator.rs
index a0d20a29bc..e5c4d7a642 100644
--- a/rs/consensus/src/consensus/validator.rs
+++ b/rs/consensus/src/consensus/validator.rs
@@ -3847,15 +3847,14 @@ pub mod test {
 
             let block = pool.make_next_block_with_rank(Rank(1));
             let mut second_block = block.clone();
-            second_block.content.as_mut().context.time += INITIAL_NOTARY_DELAY;
+            second_block.content.as_mut().context.time = Duration::from_nanos(1);
             second_block.update_content();
             let mut third_block = block.clone();
-            third_block.content.as_mut().context.time +=
-                INITIAL_NOTARY_DELAY + INITIAL_NOTARY_DELAY;
+            third_block.content.as_mut().context.time += Duration::from_nanos(2);
             third_block.update_content();
             time_source
                 .set_time(third_block.content.as_ref().context.time)
-                .ok();
+                .unwrap();
 
             pool.insert_validated(block.clone());
             pool.insert_unvalidated(second_block.clone());
@@ -3877,10 +3876,15 @@ pub mod test {
             pool.insert_unvalidated(block.clone());
             time_source
                 .set_time(block.content.as_ref().context.time)
-                .ok();
+                .unwrap();
 
             let changeset = validator.on_state_change(&PoolReader::new(&pool));
-            assert!(changeset.is_empty());
+            assert_matches!(
+                changeset[..],
+                [ChangeAction::MoveToValidated(
+                    ConsensusMessage::BlockProposal(ref proposal)
+                )] if proposal.rank() == block.rank()
+            );
         });
     }
 }
diff --git a/rs/test_utilities/artifact_pool/src/consensus_pool.rs b/rs/test_utilities/artifact_pool/src/consensus_pool.rs
index eec783f6ae..b2dc14fc98 100644
--- a/rs/test_utilities/artifact_pool/src/consensus_pool.rs
+++ b/rs/test_utilities/artifact_pool/src/consensus_pool.rs
@@ -243,13 +243,15 @@ impl TestConsensusPool {
         block.rank = rank;
         let registry_version = self.registry_client.get_latest_version();
 
-        // Increase time monotonically by at least initial_notary_delay
-        let monotonic_block_increment = self
+        // Increase time monotonically by at least one nanosecond (in case unit
+        // delay and initial notary delay are set to 0).
+        let settings = self
             .registry_client
             .get_notarization_delay_settings(self.subnet_id, registry_version)
             .unwrap()
-            .expect("subnet record should be available")
-            .initial_notary_delay
+            .expect("subnet record should be available");
+        let monotonic_block_increment = settings.initial_notary_delay
+            + settings.unit_delay * rank.0 as u32
             + core::time::Duration::from_nanos(1);
         block.context.time += monotonic_block_increment;
```

Reply:
Thanks for talking a look. I actually fixed it in a different way because this increment here and there if 1 nano_second is hard to track.